superset/assets/visualizations/catscat.js

```diff
@@ -0,0 +1,188 @@
+import d3 from 'd3';
+import d3tip from 'd3-tip';
+import { getColorFromScheme } from '../javascripts/modules/colors';
+
+import './catscat.css';
+import '../stylesheets/d3tip.css';
+
+/**
+ * computedProps
+ * returns derived properties and functions based on input.
+ * Used to conceptually distinguish the 'setting up' of
+ * a chart from the actual 'dom manipulation' to create the chart.
+ * @param {Object} props : Input properties to use to compute
+ * @returns {Object} : Object of computed props.
+ */
+function computedProps(props) {
+  const { fd, data, width, height } = props;
+
+  const padding = {
+    top: 20,
+    left: 50,
+    right: 20,
+    bottom: 50,
+  };
+
+  const plotWidth = width - (padding.right + padding.left);
+  const plotHeight = height - (padding.top + padding.bottom);
+  
+  // api: https://github.com/d3/d3-3.x-api-reference/blob/master/Quantitative-Scales.md#linear
+  const yMin = d3.min(data, d => d3.min(d.values, v => v.y));
+  const yMax = d3.max(data, d => d3.max(d.values, v => v.y));
+  const yScale = d3.scale.linear()
+    .domain([yMin, yMax])
+    .range([plotHeight, 0]);
+
+  // as this is a categorical scatterplot, our x scale will be ordinal. 
+  // api: https://github.com/d3/d3-3.x-api-reference/blob/master/Ordinal-Scales.md#ordinal_rangePoints
+  const xScale = d3.scale.ordinal()
```

Comment:
We might want xScale to be linear, too. The categories will be shown by color and shape.